lib/tests/test_lexer.c

```diff
@@ -128,53 +110,77 @@ _current_lloc(void)
 }
 
 #define assert_token_type(expected)                                     \
-  assert_gint(_current_token()->type, expected, "Bad token type at %s:%d", __FUNCTION__, __LINE__);
+  cr_assert_eq(_current_token()->type, expected, "Bad token type at %s:%d", __FUNCTION__, __LINE__);
 
 #define assert_parser_string(expected)                          \
   _next_token();                                                        \
   assert_token_type(LL_STRING);                                        \
-  assert_string(_current_token()->cptr, expected, "Bad parsed string at %s:%d", __FUNCTION__, __LINE__); \
+  cr_assert_str_eq(_current_token()->cptr, expected, "Bad parsed string at %s:%d", __FUNCTION__, __LINE__); \
 
-#define assert_parser_block(expected) \
-  _next_token();                                                        \
-  assert_token_type(LL_BLOCK);                                         \
-  assert_string(_current_token()->cptr, expected, "Bad parsed string at %s:%d", __FUNCTION__, __LINE__);
+static void
+assert_parser_block(gchar *expected)
+{
+  _next_token();
+  assert_token_type(LL_BLOCK);
+  cr_assert_str_eq(_current_token()->cptr, expected, "Bad parsed string at %s:%d", __FUNCTION__, __LINE__);
+}
 
-#define assert_parser_block_bad(parser) \
-  _next_token();                                                        \
+static void
+assert_parser_block_bad(void)
+{
+  _next_token();
   assert_token_type(LL_ERROR);
```

Comment:
This is a tricky one, because the `__FUNCTION__` and `__LINE__` does not occur directly, but in the end assert_token_type is a macro, so implicitely, these compiler variables occur here too.